tests/src/test/java/io/streamnative/pulsar/handlers/kop/TransactionWithAuthTest.java

```diff
@@ -0,0 +1,157 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.streamnative.pulsar.handlers.kop;
+
+import com.google.common.collect.Sets;
+import io.streamnative.pulsar.handlers.kop.security.oauth.OauthLoginCallbackHandler;
+import io.streamnative.pulsar.handlers.kop.security.oauth.OauthValidatorCallbackHandler;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.net.URL;
+import java.util.Collections;
+import java.util.Properties;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.pulsar.broker.authentication.AuthenticationProviderToken;
+import org.apache.pulsar.client.admin.PulsarAdmin;
+import org.apache.pulsar.client.impl.auth.oauth2.AuthenticationFactoryOAuth2;
+import org.apache.pulsar.client.impl.auth.oauth2.AuthenticationOAuth2;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import sh.ory.hydra.ApiException;
+import sh.ory.hydra.api.AdminApi;
+import sh.ory.hydra.model.OAuth2Client;
+
+@Slf4j
+public class TransactionWithAuthTest extends TransactionTest {
```

Comment:
This test can be renamed with `TransactionWithOAuthBearerAuthTest`.

But I think a better test is configuring both `plain` and `oauthbearer` mechanisms so that we can verify transaction could work for all existing supported mechanisms.

Reply:
Yes, in the current code, we have one test already covering `PLATN` mechanisms, see `SaslPlainTestBase#transactionsReadCommittedTest`.